src/CvPartner/Api/GetAllEmployees.cs

```diff
@@ -0,0 +1,46 @@
+using CvPartner.DTOs;
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using Refit;
+
+using Shared;
+
+namespace CvPartner.Api;
+
+[ApiController]
+[Route("[controller]")]
+public class GetAllEmployees : ControllerBase
+{
+    private readonly IOptionsSnapshot<AppSettings> _appSettings;
+    private readonly ILogger<GetAllEmployees> _logger;
+
+
+    public GetAllEmployees(IOptionsSnapshot<AppSettings> appSettings, ILogger<GetAllEmployees> logger)
+    {
+        _appSettings = appSettings;
+        _logger = logger;
+    }
+
+    public interface IEmployeeApi
+    {
+        [Get("/users?")]
+        Task<IEnumerable<CVPartnerUserDTO>> GetAllEmployee([Authorize("Token")] string authorization);
+    }
+
+
+    [HttpGet]
+    public async Task<IEnumerable<CVPartnerUserDTO>> Get()
+    {
+        _logger.LogInformation("Getting employees from database");
+        var cvPartnerApi = RestService.For<IEmployeeApi>("https://variant.cvpartner.com/api/v1");

```

Comment:
Selve bruken av Refit burde skjules av å ha en `CvPartnerRepository` klasse som håndterer den delen. Deretter burde man ha et Service-lag som mapper det om til vår type, og evt kaller en EmployeeService for å lagre i databasen.

Controller er dum og håndterer bare HTTP ting og kaller services.
Service er "domenelaget" som vet hva som skal hentes fra forskjellige repos og hvordan det skal sys sammen eller gjøres om på. Evt "regler" ligger også her.

Repository er også "dum" og vet bare hvor og hvordan den henter data.

`Controller -> Service -> Repository`.

Reply:
Tenker du noe sånn:

```
public class CVPartnerRepository
{
    private readonly IOptionsSnapshot<AppSettings> _appSettings;

    public CVPartnerRepository(IOptionsSnapshot<AppSettings> appSettings)
    {
        _appSettings = appSettings;
    }

    public async Task<IEnumerable<CVPartnerUserDTO>> GetCVPartnerDTO()
    {
        var cvPartnerApi = RestService.For<Interfaces.IEmployeeApi>("https://variant.cvpartner.com/api/v1");

        var employees = await cvPartnerApi.GetAllEmployee(_appSettings.Value.Token);
        return employees;
    }
}
```

som kalles i controller

```
 [HttpGet]
    public async Task<IEnumerable<CVPartnerUserDTO>> Get()
    {
        _logger.LogInformation("Getting employees from database");
        return await new CVPartnerRepository(_appSettings).GetCVPartnerDTO();
    }
```